src/app/public/modules/action-hub/action-hub.component.ts

```diff
@@ -17,30 +18,70 @@ let data: SkyActionHubData;
 export class SkyActionHubComponent {
   /**
    * Passes a SkyActionHubData object to build the action hub. The page loads until
-   * `title` has a value.
+   * `[data]` has a value.
    *
    * @param value
    */
   @Input()
   public set data(value: typeof data) {
-    this.needsAttention = value.needsAttention;
-    this.parentLink = value.parentLink;
-    this.recentLinks = value.recentLinks;
-    this.relatedLinks = value.relatedLinks;
-    this.title = value.title;
+    if (typeof value === 'undefined') {
+      this.loading = true;
+    } else {
+      this.loading = false;
+      this.needsAttention = value.needsAttention;
+      this.parentLink = value.parentLink;
+      this.recentLinks = value.recentLinks;
+      this.relatedLinks = value.relatedLinks;
```

Comment:
Thoughts on performing the related links sorting here that way the array only needs to be sorted whenever the backing data changes?

Right now the HTML can end up calling `getRecentLinksSorted()` on every change detection cycle. Also should this component be updated to use `OnPush` change detection?

Reply:
@ThomasOrtiz good call on the sort calls. I've moved them to the setter.

As far as `OnPush`, content embedded in `<sky-action-hub-buttons>` and `<sky-action-hub-content>` would be affected, and there isn't a whole lot to be gained in this scenario, so I'm disinclined to add that.

Reply:
I'd like @Blackbaud-PaulCrowder to chime in on this. We do do "OnPush" for many of our components that still embed consumer content and it is our standard for most of our components.

Reply:
After conversing with Paul, I went ahead and added `OnPush`.